backend/frontend/src/utils/webapi.js

```diff
@@ -114,8 +115,12 @@ var posts = new Map();
 var cached = null;
 function getPiazzaPosts(courseId: number){
   let endpoint = mainUrl + `/${courseId}/course/getpiazzaposts/`;
-  http.get(endpoint, function(res){
+  let ignoredata = false;
+  var req = http.get(endpoint, function(res){
     res.on('data', function(buf){
+      if(ignoredata){
```

Comment:
It specifies that we have encountered some kind of error, and that further data should be skipped